entity-framework/core/what-is-new/ef-core-10.0/whatsnew.md

```diff
@@ -21,6 +21,158 @@ EF10 requires the .NET 10 SDK to build and requires the .NET 10 runtime to run.
+            b.HasIndex(x => x.ContentsEnglish).IsFullTextIndex();
+            b.Property(x => x.ContentsGerman).EnableFullTextSearch();
+            b.HasIndex(x => x.ContentsGerman).IsFullTextIndex();
+            b.Property(x => x.TagsGerman).EnableFullTextSearch();
+            b.HasIndex(x => x.TagsGerman).IsFullTextIndex();
+        });
+    }
+```
+
+<a name="full-text-search-querying"></a>
+
+#### Querying
+
+As part of the full-text search feature, Azure Cosmos DB introduced several built-in functions which allow for efficient querying of content inside the full-text search enabled properties. These functions are: `FullTextContains`, `FullTextContainsAll`, `FullTextContainsAny`, which look for specific keyword or keywords and `FullTextScore`, which returns [BM25 score](https://en.wikipedia.org/wiki/Okapi_BM25) based on provided keywords.
+
+> [!NOTE]
+> `FullTextScore` can only be used inside `OrderBy` to rank the documents based on the score.
+
+EF Core exposes these functions as part of `EF.Functions` so they can be used in queries:
+
+```c#
+var cosmosBlogs = await context.Blogs.Where(x => EF.Functions.FullTextContainsAll(x.Contents, "database", "cosmos")).ToListAsync();
+
+var keywords = new string[] { "AI", "agent", "breakthrough" };
+var mostInteresting = await context.Blogs.OrderBy(x => EF.Functions.FullTextScore(x.Contents, keywords)).Take(5).ToListAsync();
+```
+
+<a name="full-text-search-hybrid-search"></a>
+
+#### Hybrid search
+
+Full-text search can be used with vector search in the same query (i.e. hybrid search), by combining results of `FullTextScore` and `VectorDistance` functions. It can be done using the `RRF` function (Reciprocal Rank Fusion), which EF Core also provides inside `EF.Functions`:
+
+```c#
+public class Blog
+{
+    ...
+
+    public float[] Vector { get; set; }
+    public string Contents { get; set; }
+}
+
+public class BloggingContext
+{
+    ...
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<Blog>(b =>
+        {
+            b.Property(x => x.Contents).EnableFullTextSearch();
+            b.HasIndex(x => x.Contents).IsFullTextIndex();
+
+            b.Property(x => x.Vector).IsVectorProperty(DistanceFunction.Cosine, dimensions: 1536);
+            b.HasIndex(x => x.Vector).IsVectorIndex(VectorIndexType.Flat);
+        });
+    }
+}
+
+float[] myVector = ...
+var hybrid = await context.Blogs.OrderBy(x => EF.Functions.Rrf(
+        EF.Functions.FullTextScore(x.Contents, "database"), 
+        EF.Functions.VectorDistance(x.Vector, myVector)))
+    .Take(10)
+    .ToListAsync();
+```
+
+> [!TIP]
+> You can combine more than two scoring functions inside `Rrf` call, as well as using only `FullTextScore`, or only `VectorDistance`.
+
+<a name="full-text-search-limitations"></a>
+
+#### Limitations
+
+Here are some current limitations of the full-text search using EF Core:
```

Comment:
None of the below seem like EF Core limitations, but rather Cosmos ones. BTW I wouldn't actually call these limitations, since that implies the feature is incomplete and these limitations will be removed in a future versions. I don't think this is true for these - that's just the (sensible) design of the feature.

On an unrelated note, I'd not put this here in "what's new" - that goes too deeply into detail and generally makes the "what's new" huge (we should generally strive to make it minimal, showing the feature in a relatively minimal but awesome way, and then linking to the full docs). So I think these belong in the docs section under the Cosmos provider documentation.

Reply:
Yeah, limitations doesn't sound great. Will convert these to NOTEs or something